lib/vdc/CurvilinearGrid.cpp

```diff
@@ -1028,11 +1026,11 @@ QuadTreeRectangle<float, size_t> *CurvilinearGrid::_makeQuadTreeRectangle() cons
 	const vector <size_t> dims2d = {dims[0], dims[1]};
 	size_t reserve_size = dims2d[0] * dims2d[1];
 
-	QuadTreeRectangle<float, size_t> *qtr = 
-		new QuadTreeRectangle<float, size_t>(
+    // This structure will be handed to and owned by a unique_ptr_cache,
+    // thus nobody needs to manually do the cleanup after this creation. 
+	QuadTreeRectangle<float, size_t> * qtr = new QuadTreeRectangle<float, size_t> (
```

Comment:
This doesn't seem right. Maybe I don't understand shared pointers very well. Shouldn't a unique_ptr be created here? It seems like if the caller of GetQuadTreeRectangle() doesn't destroy the returned pointer to QuadTreeRectangle it will never be freed; it's a memory leak, no?